src/solve/canonicalization.md

```diff
@@ -1,10 +1,84 @@
 # Canonicalization
 
-While the exact approach to canonicalization for this solver will differ slightly
-wrt to lifetimes, please visit [the relevant chalk chapter][chalk] for now.
+Canonicalization is the process of *isolating* a value from its context and is necessary
+for global caching of goals which include inference variables.
 
-<!-- date-check: jan 2023 -->
-As of 10 January 2023,  canonicalization is not yet fully implemented
-in the new solver.
+The idea is that given the goals `u32: Trait<?x>` and `u32: Trait<?y>`, where `?x` and `?y`
+are two different currently unconstrained inference variables, we should get the same result
+for both goals. We can therefore prove *the canonical query* `exists<T> u32: Trait<T>` once
+and reuse the result.
 
-[chalk]: https://rust-lang.github.io/chalk/book/canonical_queries/canonicalization.html#canonicalization
\ No newline at end of file
+Let's first go over the way canonical queries work and then dive into the specifics of
+how canonicalization works.
+
+## A walkthrough of canonical queries
+
+We're going to use the goal `u32: Trait<?x>` as an example and assume that this only holds
+for `u32: Trait<Vec<?z>>` where `?z` is unconstrained.
```

Comment:
Would it maybe make sense to say "this only holds for `for<T> u32: Trait<Vec<T>>`" or something to make it clear that its not that there is only a trait impl _specifically_ for `?z` but that it holds for any type argument to `Vec`. Or maybe even just directly say "assumes there is only an `impl<T> Trait<Vec<T>> for u32 {`

Reply:
done